storage/range.go

```diff
@@ -387,11 +387,16 @@ func (r *Range) redirectOnOrAcquireLeaderLease(timestamp proto.Timestamp) error
 	return nil
 }
 
-// verifyLeaderLease checks whether the requesting replica (by raft
-// node ID) holds the leader lease covering the specified timestamp.
-func (r *Range) verifyLeaderLease(originRaftNodeID proto.RaftNodeID, timestamp proto.Timestamp) bool {
-	l := r.getLease()
-	return originRaftNodeID == l.RaftNodeID && timestamp.Less(l.Expiration)
+// verifyLeaderLease checks whether the requesting replica (by raft node ID)
+// holds the leader lease covering the specified timestamp.
+func verifyLeaderLease(lease *proto.Lease, replicaRaftNodeID proto.RaftNodeID, timestamp proto.Timestamp) (bool, bool) {
```

Comment:
This method is pretty weird. It has two callers:
- `processRaftCommand`, which only cares if the lease is live for the supplied node, and
- `HasLeaderLease`, which has a few callers:
  - `redirectOnOrAcquireLeaderLease, which only cares if the lease is live for another range,
  - `shouldQueue` which only cares if the lease is live for **any** range,
  - `updateStoreStatus`, which only checks if the lease is live for **any** range, but seems like it should check if the lease is live for the caller.

I think this API is super clumsy. Can we just have two methods?
`func (l *lease) Live(timestamp Timestamp) bool` returns true if the lease is unexpired.
`func (r *Range) HasLiveLeaderLease(timestamp proto.Timestamp) bool` which returns true if the lease is live and held by this range.